src/core/deployer.sol

```diff
@@ -12,187 +11,68 @@
     }
-}
-
-contract Deployer {
-    TitleFab          titlefab;
-    LightSwitchFab    lightswitchfab;
-    ShelfFab          shelffab;
-    DistributorFab    distributorFab;
-    PileFab           pilefab;
-    PrincipalFab      principalFab;
-    CollectorFab      collectorFab;
-    ThresholdFab      thresholdFab;
-
-    address     public god;
-
-    Title       public title;
-    LightSwitch public lightswitch;
-    Shelf       public shelf;
-    Distributor  public distributor;
-    LenderLike  public lender;
-    Pile        public pile;
-    Principal   public principal;
-    Collector   public collector;
-    PushRegistry public threshold;
 
+    function wireDeployment() public auth {
+        require(address(borrowerDeployer) != address(0));
+        require(address(lenderDeployer) != address(0));
 
-    constructor (address god_, TitleFab titlefab_, LightSwitchFab lightswitchfab_, ShelfFab shelffab_, DistributorFab distributorFab_, PileFab pilefab_, PrincipalFab principalFab_, CollectorFab collectorFab_, ThresholdFab thresholdFab_) public {
-        god = god_;
 
-        titlefab = titlefab_;
-        lightswitchfab = lightswitchfab_;
-        shelffab = shelffab_;
-        distributorFab = distributorFab_;
-        pilefab = pilefab_;
-        principalFab = principalFab_;
-        collectorFab = collectorFab_;
-        thresholdFab = thresholdFab_;
-    }
-
-    function deployThreshold() public {
-        threshold = thresholdFab.newThreshold();
-        threshold.rely(god);
-
-    }
-    function deployCollector() public {
-        collector = collectorFab.newCollector(address(distributor), address(shelf), address(pile), address(threshold));
-        collector.rely(god);
-    }
-
-    function deployPile() public {
-        pile = pilefab.newPile();
-        pile.rely(god);
-    }
-
-    function deployTitle(string memory name, string memory symbol) public {
-        title = titlefab.newTitle(name, symbol);
-        title.rely(god);
-    }
-
-    function deployLightSwitch() public {
-        lightswitch = lightswitchfab.newLightSwitch();
-        lightswitch.rely(god);
-    }
+        // distributor allowed to call
+        borrowerDeployer.shelf().rely(address(lenderDeployer.distributor()));
 
-    function deployShelf(address currency_) public {
-        shelf = shelffab.newShelf(currency_, address(title), address(pile), address(principal));
-        shelf.rely(god);
     }
 
-    // note: this method will be refactored with the new lender side contracts, we will rely on God once more
-    //and the Pile should articulate that it depends on the distributor, not a generic "lender".
-    function deployDistributor(address currency_) public {
-        distributor = distributorFab.newDistributor(address(shelf), currency_);
-        shelf.depend("lender", address(distributor));
+    function denyBorrowAdmin(address usr) public auth {
+        borrowerDeployer.principal().deny(usr);
```

Comment:
what about threshold and collector

Reply:
yes, I only added the once which are currently need to fix the tests.

I will add the rest